### PR TITLE
Expose deploy status via etcd

### PR DESCRIPTION
**Pull request by ynachiket**
<h2> Why </h2>

We need a way to poll a channel and check if it is deployed with the latest tag. This PR is to expose the deployment status via etcd.

### CODE REVIEW

**Comment by ynachiket** [comment]
@rphillips used the setSync as I do not want this to be blocking.

**Comment by rphillips** [comment]
there is more than one upgrade service that runs. This will race with the other ones. I sorta think they should register as `/deploys/status/[channel]/[VUS instance]`. On a complete deploy, then delete the [VUS instance] node on completion... maybe TTL it for 2 hours to delete it automatically.

**Comment by ynachiket** [comment]
@rphillips I see what you are saying.

Alternate thought: How about we do it https://github.com/virgo-agent-toolkit/virgo-update-service/blob/expose_status_through_etcd/server/lib/deploy/index.js#L97 , that way we will also be consistent with the http api which has the current_downloads field? 

example /deploys/status

`{
  "success": true,
  "error": null,
  "values": {
    "current_downloads": [
      "1.1.0-81"
    ],
    "current_deploys": {
      "test": {
        "version": "1.1.0-81",
        "channel": "test"
      }
    }
  }
} `

and once the download is complete the field is empty.

Thoughts?

**Comment by rphillips** [comment]
that would work too

File: server/lib/deploy/index.js [hunk 1]
```diff
@@ -89,12 +89,16 @@ Deploy.prototype._download = function(version, callback) {
     return;
   }
   CURRENT_DOWNLOADS[version] = true;
+  etcdClient = new etcd.Client(self.options.etcd_host, self.options.etcd_port);
+  etcdClient.setSync('/deploys/downloads', JSON.stringify({"version": version, "status": "Downloading"}), {ttl: 3600});
```
Comment (ynachiket) [inline server/lib/deploy/index.js]: @rphillips 
Approach: set version and download status and let it live for an hour , as soon as download is complete set status to downloaded for that version.

**Comment by rphillips** [comment]
ok. that works +1

**Pull request merged**